app/waf/src/models/as3.model.ts

```diff
@@ -1,387 +1,47 @@
-import {
-  Adc,
-  Application,
-  Service,
-  Pool,
-  Member,
-  Wafpolicy,
-  Endpointpolicy,
-  Rule,
-  Action,
-  Condition,
-} from '.';
-import {isNullOrUndefined} from 'util';
+import {Adc, Application, Declaration} from '.';
 
-// TODO: use json-schema-to-ts re-implement the following?
-// Or change to type definition?
-// as3-schema definition:
-//  https://github.com/F5Networks/f5-appsvcs-extension/blob/master/schema/latest/as3-schema-3.9.0-3.json
-// json-schema-to-typescript
-//  https://www.npmjs.com/package/json-schema-to-typescript
-export class AS3Object extends Object {
-  class: string;
-
-  constructor() {
-    super();
-  }
-}
-
-export class AS3JSONObject extends Object {
-  [key: string]: Object;
+export class AS3Declaration {
+  [key: string]: undefined | string | number | boolean | object;
 }
 
-export class AS3DeployRequest extends AS3Object {
-  action: string;
+export class AS3DeployRequest {
+  readonly class: string = 'AS3';
+  readonly action: 'deploy';
   targetHost: string;
   targetPort: number;
   targetUsername: string;
   targetPassphrase: string;
   declaration: AS3Declaration;
 
-  constructor(params: {[key: string]: Object}) {
-    super();
-
-    let adc = <Adc>params['adc'];
-
-    this.class = 'AS3';
-    this.action = 'deploy';
+  constructor(adc: Adc, application: Application, declaration?: Declaration) {
```

Comment:
Again, I think `constructor(adc: Adc, declaration?: Declaration) {` is better.

Reply:
Some information of AS3 request if from Application resource, for example, the application name.